src/measure_theory/l1_space.lean

```diff
@@ -0,0 +1,200 @@
+  neg_mem := λ _, integrable_neg }
+
+section normed_space
+variables {K : Type*} [normed_field K] [second_countable_topology K] [normed_space K γ]
+
+lemma integrable_smul : ∀ {c : K} {f : α →ₘ γ}, integrable f → integrable (c • f) :=
+by { assume c, rintros ⟨f, hf⟩, have := integrable_smul hf, simpa }
+
+end normed_space -- section
+
+end ae_eq_fun
+
+section
+variables (α γ)
+def l1 : Type* := subtype (@ae_eq_fun.integrable α _ γ _ _)
+
+local notation `L¹` := l1
+
+infixr ` →₁ `:25 := l1
+
+end
+
+namespace l1
+open ae_eq_fun
+
+section normed_group
+
+def mk (f : α → γ) : measurable f → integrable f → (α →₁ γ) :=
+assume hfm hfi, ⟨mk f hfm, by { rw integrable_mk, assumption }⟩
+
+@[simp] protected lemma eq {f g : α →₁ γ} : f = g ↔ f.1 = g.1 := ⟨congr rfl , subtype.eq⟩
+
+instance : emetric_space (α →₁ γ) := subtype.emetric_space
+instance : metric_space (α →₁ γ) := metric_space_emetric_ball 0 ⊤
+
+@[simp] lemma dist_def (f g : α →₁ γ) : dist f g = ennreal.to_real (edist f.1 g.1) := rfl
+
+instance : add_comm_group (α →₁ γ) := subtype.add_comm_group
+
+@[simp] lemma zero_def : (0 : α →₁ γ) = ⟨(0 : α →ₘ γ), ae_eq_fun.integrable_zero⟩ := rfl
+
+@[simp] lemma add_def (f g : α →₁ γ) : f + g = ⟨f.1 + g.1, ae_eq_fun.integrable_add f.2 g.2⟩ := rfl
+
+instance : has_norm (α →₁ γ) := ⟨λ f, dist f 0⟩
+
+@[simp] lemma norm_def (f : α →₁ γ) : (norm f) = ennreal.to_real (edist f.1 0) := rfl
+
+instance : normed_group (α →₁ γ) := normed_group.of_add_dist (λ x, rfl) $ by
+{ rintros ⟨f, _⟩ ⟨g, _⟩ ⟨h, _⟩, simp only [dist_def, add_def], rw [edist_eq_add_add] }
+
+end normed_group -- section
+
+section normed_space
+
+variables {K : Type*} [normed_field K] [second_countable_topology K] [normed_space K γ]
+
+protected def smul : K → (α →₁ γ) → (α →₁ γ) := λ x f, ⟨x • f.1, ae_eq_fun.integrable_smul f.2⟩
+
+instance : has_scalar K (α →₁ γ) := ⟨l1.smul⟩
+
+@[simp] lemma smul_def {x : K} {f : α →₁ γ} : x • f = ⟨x • f.1, ae_eq_fun.integrable_smul f.2⟩ := rfl
+
+protected lemma one_smul : ∀ f : α →₁ γ, (1 : K) • f = f :=
+by { rintros ⟨f, hf⟩, simp [ae_eq_fun.one_smul f] }
+
+protected lemma mul_smul : ∀ (x y : K) (f : α →₁ γ), (x * y) • f = x • y • f :=
+by { rintros x y ⟨f, hf⟩, simp [ae_eq_fun.mul_smul x y f] }
+
+protected lemma smul_add : ∀ (x : K) (f g : α →₁ γ), x • (f + g) = x • f + x • g :=
+by { rintros x ⟨f, hf⟩ ⟨g, hg⟩, simp [smul_add x f g] }
+
+protected lemma smul_zero : ∀ (x : K), x • (0 : α →₁ γ) = 0 :=
+by { assume x, simp [smul_zero x] }
+
+protected lemma add_smul : ∀ (x y : K) (f : α →₁ γ), (x + y) • f = x • f + y • f :=
+by { rintros x y ⟨f, hf⟩, simp [add_smul x y f] }
+
+protected lemma zero_smul : ∀ f : α →₁ γ, (0 : K) • f = 0 :=
+by { rintro ⟨f, hf⟩, simp [zero_smul K f] }
+
+protected lemma norm_smul : ∀ (x : K) (f : α →₁ γ), norm (x • f) = norm x * norm f :=
+begin
+  rintros x ⟨f, hf⟩,
+  show ennreal.to_real (edist (x • f) 0) = ∥x∥ * ennreal.to_real (edist f 0),
+  rw [edist_smul, to_real_of_real_mul], exact norm_nonneg _
+end
+
```

Comment:
all these protected lemmas are note necessary to be given a name. Can you move them into the definition of `instance : semimodule...`